src/lib/components/event/event-summary.svelte

```diff
@@ -1,49 +1,104 @@
   import EventEmptyRow from './event-empty-row.svelte';
-  import Loading from '$lib/holocene/loading.svelte';
-  import { updating } from '$lib/stores/events';
+  import ApiPagination from '$lib/holocene/api-pagination.svelte';
+  import { groupEvents } from '$lib/models/event-groups';
+  import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
+  import KeyboardShortcut from '$lib/holocene/keyboard-shortcut/shortcut.svelte';
+  import Tooltip from '$lib/holocene/tooltip.svelte';
+  import { authUser } from '$lib/stores/auth-user';
+  import EventShortcutKeys from './event-shortcut-keys.svelte';
 
-  export let items: IterableEvents;
-  export let groups: EventGroups;
-  export let loading = false;
   export let compact = false;
+  let showShortcuts = false;
 
   function handleExpandChange(event: CustomEvent) {
     $expandAllEvents = event.detail.expanded;
   }
+
+  $: category = $page.url.searchParams.get('category');
+  $: total = category ? '' : $eventHistory.end[0]?.id;
+
+  $: fetchEvents = () => {
+    return getPaginatedEvents({
+      namespace: $page.params.namespace,
+      workflowId: $page.params.workflow,
+      runId: $page.params.run,
+      sort: $eventFilterSort,
+      category,
+      compact,
+      settings: $page.stuff.settings,
+      accessToken: $authUser?.accessToken,
+    });
+  };
+
+  const onShiftUp = (event: KeyboardEvent) => {
+    if (!compact) {
+      const sort = 'ascending';
+      $eventFilterSort = sort;
+      updateQueryParameters({
+        parameter: 'sort',
+        value: 'ascending',
+        url: $page.url,
+      });
+    }
+  };
+
+  const onShiftDown = (event: KeyboardEvent) => {
+    if (!compact) {
+      const sort = 'descending';
+      $eventFilterSort = sort;
+      updateQueryParameters({
+        parameter: 'sort',
+        value: sort,
+        url: $page.url,
+      });
+    }
+  };
+
+  const onSpace = (event: KeyboardEvent) => {
+    event.preventDefault();
+    showShortcuts = !showShortcuts;
+  };
 </script>
 
-{#if loading}
-  <Loading />
-{:else}
-  <Pagination
-    {items}
-    floatId="event-view-toggle"
+<EventShortcutKeys open={showShortcuts} {compact} />
+{#key [$eventFilterSort, category, $refresh]}
+  <ApiPagination
     let:visibleItems
-    let:initialItem
+    let:updating
+    let:activeIndex
+    let:setActiveIndex
+    onFetch={fetchEvents}
+    onError={(error) => console.error(error)}
+    pageSizeOptions={[]}
+    {onShiftUp}
+    {onShiftDown}
+    {onSpace}
+    {total}
   >
-    <EventSummaryTable
-      updating={$updating}
-      {compact}
-      on:expandAll={handleExpandChange}
-    >
-      {#each visibleItems as event (`${event.id}-${event.timestamp}`)}
+    <EventSummaryTable {updating} {compact} on:expandAll={handleExpandChange}>
+      {@const events = compact ? groupEvents(visibleItems) : visibleItems}
+      {#each events as event, index (`${event.id}-${event.timestamp}`)}
```

Comment:
Should this be something we do on the model when we generate it? Even if we call it like `idWithTimestap` or something like that. Do we only want the `timestamp` to make it unique? If so, maybe we should just use `uuid` instead in the model.